### ISSUE TITLE
Bug: Unsubscribing from DiagnosticSource events does not work

### ISSUE THREAD

**Issue by gregkalapos** [elastic/apm-agent-dotnet#72]
This is more pressuring for writing tests.

Currently code like this does not unsubscribe the agent from receiving events and creating spans:

```
using (agent.Subscribe(new HttpDiagnosticsSubscriber()))
{
	//do stuff			
}
//we should not be subscribed here, but we are.
```

Therefore subsequent tests that assume the `HttpDiagnosticsSubscriber` is not registered will fail.

**Comment by gregkalapos** [comment]
The root cause was in [DiagnosticInitializer.cs](https://github.com/elastic/apm-agent-dotnet/blob/master/src/Elastic.Apm/DiagnosticSource/DiagnosticInitializer.cs ):

```
public void OnNext(DiagnosticListener value)
{
   foreach (var listener in _listeners)
   {
      if (value.Name == listener.Name) value.Subscribe(listener);
   }
}
```

`value.Subscribe` returns `IDisposable` which wasn't disposed. Fix is on the way.

**Comment by gregkalapos** [comment]
Fixed by #75
